Libraries/libutil/Sources/Relative.cpp

```diff
@@ -40,56 +40,90 @@ NormalizePath(
     char separator,
     bool(*isSeparator)(char c))
 {
-    std::string output;
-
-    for (auto it = start; it != end; ++it) {
-        if (isSeparator(*it)) {
-            /* Skip runs of multiple separators. */
-            while (std::next(it) != end && isSeparator(*std::next(it))) {
-                ++it;
-            }
-
-            /* Only after the beginning, end, or after an existing trailing separator. */
-            if (!output.empty() && std::next(it) != end && !isSeparator(output.back())) {
-                output += separator;
-            }
-        } else if (collapse && (it == start || isSeparator(*std::prev(it))) && *it == '.') {
-            /* Handle relative paths. */
-            if (std::next(it) == end || isSeparator(*std::next(it))) {
-                /* Path component is current. */
-                if (!output.empty() && isSeparator(output.back())) {
+  std::vector<std::string> output;
+  std::string component;
+
+  for (auto it = start; it != end; ++it) {
```

Comment:
It looks like you screwed up the indentation. Undo these whitespace changes?

Reply:
Sure.